### PR TITLE
Add ValidComponentChildren.toArray

### PR DESCRIPTION
**Pull request by taion**
This replaces the inconsistent childrenToArray.

### CODE REVIEW

File: src/utils/ValidComponentChildren.js [hunk 1]
```diff
@@ -137,11 +135,26 @@ function some(children, func, context) {
   return result;
 }
 
+function toArray(children) {
```
Comment (jquense) [inline src/utils/ValidComponentChildren.js]: does it make sense to leverage `React.Children.toArray` here now that it exists?

Comment (taion) [inline src/utils/ValidComponentChildren.js]: Like `React.Children.toArray(children).filter(React.isValidElement)`? I feel like they're about equivalent.

Comment (jquense) [inline src/utils/ValidComponentChildren.js]: Yeah I don't know, toArray does some bit of logic, but we may already have it covered

Comment (taion) [inline src/utils/ValidComponentChildren.js]: Not really – it's just `map` with the identity function as the predicate: https://github.com/facebook/react/blob/24dfb56113a214d98f29f69e2c26f67d7e947067/src/isomorphic/children/ReactChildren.js#L196-L205

**Pull request merged**